### PR TITLE
Rename template config and move transformations to code snippets

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: docs/snippets/CMakeLists.txt [hunk 1]
```diff
@@ -61,6 +61,7 @@ target_include_directories(${TARGET_NAME} PRIVATE "${OpenVINO_SOURCE_DIR}/src/in
                                                   "${OpenVINO_SOURCE_DIR}/src/common/low_precision_transformations/include"
                                                   "${OpenVINO_SOURCE_DIR}/src/frontends/common/include"
                                                   "${OpenVINO_SOURCE_DIR}/src/core/template_extension/new/")
+ov_mark_target_as_cc(${TARGET_NAME})
```
Comment (ilya-lavrenov) [inline docs/snippets/CMakeLists.txt:64]: why is it required?

File: docs/Extensibility_UG/graph_rewrite_pass.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
 `ov::pass::GraphRewrite` serves for running multiple matcher passes on `ov::Model` in a single graph traversal.
 Example:
 
-@snippet src/transformations/template_pattern_transformation.cpp matcher_pass:graph_rewrite
+@snippet template_pattern_transformation.cpp matcher_pass:graph_rewrite
```
Comment (ilya-lavrenov) [inline docs/Extensibility_UG/graph_rewrite_pass.md:6]: why do we need this move?

Comment (ilyachur) [reply docs/Extensibility_UG/graph_rewrite_pass.md:6]: These transformation doesn't provide any benefits for template plugin. So I just moved them to snippets

File: docs/snippets/CMakeLists.txt (hunk 1 above)
Comment (ilyachur) [reply docs/snippets/CMakeLists.txt:64]: Transformations contain CC macro

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**